index.js

```diff
@@ -128,6 +130,18 @@ class TuyaDevice extends EventEmitter {
       try {
         // Send request
         this._send(buffer).then(async data => {
```

Comment:
Do we want to wrap get() in a queue too?

Reply:
I don't think there's any reason to wrap get() in a queue, as it doesn't suffer from the same issues as set() as it uses sequence numbers and adds the resolver to an array based on this sequence number.  The packet handler then resolves the correct send() based on the received sequence number so we can be assured that the response DP_QUERY returned by send() is properly associated to the request DP_QUERY and multiple outstanding requeues are no issue.  Also, send() should timeout after 5 retries to send the DP_QUERY, so I believe get() will properly timeout in most cases although there may still be a hang case where send() somehow to somehow send DP_QUERY, but no response DP_QUERY is received by the packet handler, so maybe an upper level timeout is a good idea.

However, with set(), a CONTROL command is used for send(), and indeed send() does received a CONTROL response with sequence number, but that CONTROL response is basically just an acknowledgement that the set happened so the code doesn't use the CONTROL response to resolve the set().  Instead it waits for the next STATUS message from the device and uses this to resolve the set().  The problem with this is that STATUS is not technically a response to CONTROL, the device just sends STATUS because something changed, so the STATUS command has no sequence number and thus there's no way to tie a specific STATUS to a specific CONTROL.

The code as it exist today just assumes that the next received STATUS is a response to the most recently sent CONTROL, which is reasonable, as long as there aren't multiple set() running concurrently or one of the sets doesn't get a response, which causes it to hang forever.  The only way to be reasonably sure that the next STATUS is a response to the CONTROL is to have only one set() at a time, this is especially true if you issue a CONTROL to a non-exist DPS, since the device will not acknowledge that with a STATUS message at all.

Now, with a little research, I may be able to figure out how to detect a CONTROL response to a non-exist DPS as I'm guessing the device probably returns an error in the CONTROL response.  Then we wouldn't have to rely on the timeout for that case. but I haven't gone down that path yet as my goal was just to try to make it work without hanging as it does now.